kapitan/targets.py

```diff
@@ -288,145 +287,27 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, **kwa
         if input_type == "jinja2":
-            _compile_path = os.path.join(compile_path, target_name, output_path)
-            # support writing to an already existent dir
-            try:
-                os.makedirs(_compile_path)
-            except OSError as ex:
-                # If directory exists, pass
-                if ex.errno == errno.EEXIST:
-                    pass
-            for input_path in input_paths:
-                jinja2_file_found = False
-                for path in search_paths:
-                    compile_path_sp = os.path.join(path, input_path)
-                    if os.path.exists(compile_path_sp):
-                        jinja2_file_found = True
-                        # copy ext_vars to dedicated jinja2 context so we can update it
-                        ctx = ext_vars.copy()
-                        ctx["inventory"] = inventory(search_paths, target_name)
-                        ctx["inventory_global"] = inventory(search_paths, None)
-                        try:
-                            compile_jinja2(compile_path_sp, ctx, _compile_path, ref_controller,
-                                           target_name=target_name, **kwargs)
-                        except KapitanError as e:
-                            raise CompileError("{}\nCompile error: failed to compile target: {}".format(e, target_name))
-
-                if not jinja2_file_found:
-                    raise CompileError("Compile error: {} for target: {} not found in "
-                                       "search_paths: {}".format(input_path, target_name, search_paths))
-
-    logger.info("Compiled %s (%.2fs)", target_name, time.time() - start)
-
-
-def compile_jinja2(path, context, compile_path, ref_controller, **kwargs):
-    """
-    Write items in path as jinja2 rendered files to compile_path.
-    path can be either a file or directory.
-    kwargs:
-        reveal: default False, set to reveal refs on compile
-        target_name: default None, set to current target being compiled
-    """
-    reveal = kwargs.get('reveal', False)
-    target_name = kwargs.get('target_name', None)
-
-    for item_key, item_value in render_jinja2(path, context).items():
-        full_item_path = os.path.join(compile_path, item_key)
-        try:
-            os.makedirs(os.path.dirname(full_item_path))
-        except OSError as ex:
-            # If directory exists, pass
-            if ex.errno == errno.EEXIST:
-                pass
-        with CompiledFile(full_item_path, ref_controller, mode="w", reveal=reveal, target_name=target_name) as fp:
-            fp.write(item_value["content"])
-            mode = item_value["mode"]
-            os.chmod(full_item_path, mode)
-            logger.debug("Wrote %s with mode %.4o", full_item_path, mode)
+            input_compiler = jinja2_compiler
+        elif input_type == "jsonnet":
+            input_compiler = jsonnet_compiler
+        else:
+            raise CompileError("")
```

Comment:
We should print that the input type is not valid, and which ones we support

Reply:
Oops, forgot that. Will fix.

Reply:
done